lambdas/data-migration2/src/pdrs.ts

```diff
@@ -0,0 +1,136 @@
+import Knex from 'knex';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import Logger from '@cumulus/logger';
+import {
+  getRecordCumulusId,
+  PostgresCollectionRecord,
+  PostgresExecutionRecord,
+  PostgresProviderRecord,
+  PostgresPdrRecord,
+  PostgresPdr,
+  tableNames,
+} from '@cumulus/db';
+import { envUtils } from '@cumulus/common';
+import { RecordAlreadyMigrated } from '@cumulus/errors';
+
+import { MigrationSummary } from './types';
+
+const logger = new Logger({ sender: '@cumulus/data-migration/pdrs' });
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+
+/**
+ * Migrate PDR record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordAlreadyMigrated} if record was already migrated
+ */
+export const migratePdrRecord = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<void> => {
+  // Validate record before processing using API model schema
+  Manager.recordIsValid(dynamoRecord, schemas.pdr);
+
+  const existingRecord = await knex<PostgresPdrRecord>(tableNames.pdrs)
+    .where({
+      name: dynamoRecord.pdrName,
+    })
+    .first();
+
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`PDR name ${dynamoRecord.pdrName} was already migrated, skipping.`);
+  }
+
+  const collectionCumulusId = await getRecordCumulusId<PostgresCollectionRecord>(
+    { name: dynamoRecord.collectionId },
+    tableNames.collections,
+    knex
+  );
+
+  const providerCumulusId = await getRecordCumulusId<PostgresProviderRecord>(
```

Comment:
Likewise, I'd like to replace these with `collectionPgModel.getRecordCumulusId` and `providerPgModel.getRecordCumulusId` respectively